File: frontend/AngularPcMatchUp/src/app/features/order/interfaces/order.dto.ts
import { AddressDto } from '../../address/interfaces/address.dto';
import { OrderDetailDto } from './order-detail.dto';

export interface OrderDto {
  orderId: number;
  status: string;
  total: number;
  createdAt: string;
  address: AddressDto;
  details: OrderDetailDto[];
}

File: backend/APIEccomerce/Models/DTOs/OrderDto.cs
using System.Text.Json.Serialization;

namespace APIEccomerce.Models.DTOs
{
    public class OrderDto
    {
        [JsonPropertyName("orderId")]
        public int OrderId { get; set; }

        [JsonPropertyName("status")]
        public string Status { get; set; } = string.Empty;

        [JsonPropertyName("total")]
        public decimal Total { get; set; }

        [JsonPropertyName("createdAt")]
        public DateTime CreatedAt { get; set; }

        [JsonPropertyName("address")]
        public AddressDto Address { get; set; } = null!;

        [JsonPropertyName("details")]
        public List<OrderDetailDto> Details { get; set; } = new();
    }

    
}
